### DISCUSSION TITLE
Dependencies of CI/CD on Windows platform

### DISCUSSION THREAD

**Discussion by bhashemian** [Project-MONAI/MONAI#1786] [General]
**Describe the bug**
Right now CI/CD is set to install the first 12 dependencies for Windows platform, so it excludes any new dependency. Shouldn't we specifically exclude the ones that are not running on Windows instead?

**Comment by bhashemian** [comment]
@wyli @Nic-Ma @rijobro 
Do you know what are the troublesome dependencies for Windows platform that need to be excluded?

**Comment by bhashemian** [comment]
Is there any reason we are relying on requirement files instead of `setup.cfg`?

**Reply by Nic-Ma** [reply]
Hi @wyli ,

Could you please help share some comments here as you developed the CI / CD files before, I don't remember clearly.

Thanks in advance.

**Reply by wyli** [reply]
`requirements.txt` seems to work nicely with the `pip` command, and we can't fully rely on `setup.cfg` only because there are modern methods such as pep518 https://www.python.org/dev/peps/pep-0518/

**Comment by wyli** [comment]
most of the deps declare support for windows, but in practice it's difficult to ensure consistent cross-platform behaviour from our side. currently we have the preliminary tests to cover the first 12 essential packages. the skipped ones are just code type/style checking on windows, which are not really essential for us
